### ISSUE TITLE
Wrong ppc64le busybox binary in base:debug image.

### ISSUE THREAD

**Issue by Siddhesh-Ghadi** [GoogleContainerTools/distroless#716]
The `base:debug` image fails with `exec error` on `ppc64le` architecture due to wrong busybox binary. The failure is described in https://github.com/GoogleContainerTools/distroless/pull/622#issuecomment-795077067. Image build picks up busybox binaries for all architectures from https://busybox.net/downloads/binaries/ . Ppc64le binaries which are present herein are for big endian and busybox community is unwilling to provide little endian(LE) binaries.(https://bugs.busybox.net/show_bug.cgi?id=13491)

The resolution is to use busybox ppc64le binary provided by [docker-library/busybox](https://github.com/docker-library/busybox/blob/dist-ppc64le/stable/musl/busybox.tar.xz).

This issue is a blocker for tektoncd on ppc64le (https://github.com/tektoncd/pipeline/issues/3820).

**Comment by chanseokoh** [comment]
I think distroless can just install the busybox package that comes with the Debian distro in same usual way other it installs packages Debian packages. I guess the reason we didn't do it was because the version included in the Debian 9 distro was too old?

**Comment by Siddhesh-Ghadi** [comment]
The debian version is to old and create issues for some of the test cases in tektoncd. On the other hand, the busybox binary provided by `docker-library` is latest and I have managed to successfully add it in `base:debug` image for `ppc64le` by making minor changes in [experimental/busybox/BUILD](https://github.com/GoogleContainerTools/distroless/blob/master/experimental/busybox/BUILD) file.

**Comment by chanseokoh** [comment]
Looks like you are using `gcr.io/distroless/base`, which was Debian 9-based until yesterday. It has switched to Debian 10, so I think it should have a newer version. Can you re-do the test with Debian 10?

**Comment by Siddhesh-Ghadi** [comment]
Doesn't work.
- gcr.io/distroless/base:debug
```
$ docker run -it --rm gcr.io/distroless/base:debug
standard_init_linux.go:211: exec user process caused "exec format error"
```
- gcr.io/distroless/base-debian10:debug
```
docker run -it --rm gcr.io/distroless/base-debian10:debug
standard_init_linux.go:211: exec user process caused "exec format error"
```

I think all debug images are packing faulty busybox binary for `ppc64le`.
https://github.com/GoogleContainerTools/distroless/blob/b6b2611dbbd59f7b5c806e3008973869dffcac2f/WORKSPACE#L281

**Comment by chanseokoh** [comment]
Sorry, I was talking about testing with the busybox package that comes with Debian 10, since you said in your test, the Debian 9 busybox was found to be too old.

**Comment by chanseokoh** [comment]
Looks like the Debian 10 busybox version will be [1.30.1](https://packages.debian.org/buster/busybox), whereas the Debian 9 is [1.22.0](https://packages.debian.org/stretch/busybox). I think it's better to use the Debian busybox package unless there is a good reason.

**Comment by Siddhesh-Ghadi** [comment]
Before testing with busybox from `docker-library/busybox`, I tried [1.30.1-4](https://packages.debian.org/buster/busybox) & [1.30.1-6](https://packages.debian.org/sid/busybox) from debian repos, none of them worked for tekton testing. The minimum required busybox version is `1.31` which is currently installed in `arm`, `amd` & `s390x` images correctly.

**Comment by dlorenc** [comment]
If you want to change the http_file in the WORKSPACE to point to whatever version you want it should be fine.

**Comment by Siddhesh-Ghadi** [comment]
I am planning to point `http_file` for ppc64le to [busybox.tar.xz](https://github.com/docker-library/busybox/blob/dist-ppc64le/stable/musl/busybox.tar.xz) which is the latest version available for `ppc64le`. Along with this, since it is a tarball, some preprocessing is required using if condition in https://github.com/GoogleContainerTools/distroless/blob/b6b2611dbbd59f7b5c806e3008973869dffcac2f/experimental/busybox/BUILD#L11-L16

**Comment by Siddhesh-Ghadi** [comment]
Resolved with #723
